test/issues/340.js

```diff
@@ -0,0 +1,149 @@
+// Copyright © 2018 IBM Corp. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* global describe it before beforeEach after */
+'use strict';
+
+const assert = require('assert');
```

Comment:
Can't we just add these new tests to the `plugins/iamauth.js` and avoid all the duplication of setup tokens etc. Having recently gone through adding the `SERVER_WITH_CREDS` stuff to allow us to test in more envs I can tell you that having this stuff in fewer places is a good thing!